### DISCUSSION TITLE
"sideEffects" is false in the enclosing "package.json" file warning

### DISCUSSION THREAD

**Discussion by wiesson** [sveltejs/kit#3630] [Q&A]
I have this output messages
```.svelte-kit/output/server/entries/pages/companies/_id_/events/create.svelte.js:7:7: warning: Ignoring this import because "node_modules/@supabase/supabase-js/dist/main/index.js" was marked as having no side effects
     7 │ import "@supabase/supabase-js";
       ╵        ~~~~~~~~~~~~~~~~~~~~~~~
   node_modules/@supabase/supabase-js/package.json:21:2: note: "sideEffects" is false in the enclosing "package.json" file
    21 │   "sideEffects": false,
       ╵   ~~~~~~~~~~~~~

.svelte-kit/output/server/chunks/utils-f1fef778.js:2:7: warning: Ignoring this import because "node_modules/lodash-es/lodash.js" was marked as having no side effects
     2 │ import "lodash-es";
       ╵        ~~~~~~~~~~~
   node_modules/lodash-es/package.json:14:2: note: "sideEffects" is false in the enclosing "package.json" file
    14 │   "sideEffects": false,
       ╵   ~~~~~~~~~~~~~
```

Does anyone know what I should do here? I expect that those packages have to add" sideEffects: false" to their package.json in order to fix it?

**Comment by bluwy** [comment]
You can ignore them. It is because Vite's build (or Rollup) preserve externalized imports as `import 'blabla'` regardless if it has `"sideEffects": false` denoted, which usually means you can strip the imports away (don't preserve it). This is generally harmless, but certainly incurs a slight overhead in runtime. I've been meaning to check this out on Vite's side but it's not much a priority now.

Note: The output messages comes from esbuild, which happens on the adapt stage, after Vite build.
